File: frontend/src/types/diagnostic.ts
export interface DiagnosticSession {
  id: string;
  asset_id?: string;
  technician_id?: string;
  user_question: string;
  image_id?: string;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  created_at: string;
  completed_at?: string;
  asset?: {
    id: string;
    name: string;
    asset_code?: string;
    model?: string;
  };
  technician?: {
    full_name: string;
  };
  results?: DiagnosticResult[];
}

export interface DiagnosticResult {
  id: string;
  diagnostic_session_id: string;
  issue_summary: string;
  identified_error_code?: string;
  confidence?: number;
  probable_causes: Array<{ cause: string; probability?: number }>;
  recommended_steps: Array<{ step: number; action: string; safety_note?: string }>;
  required_tools: string[];
  safety_warnings: string[];
  citations: Array<{ document_title: string; page_number: number; chunk_id?: string }>;
  model_name?: string;
  response_time_ms?: number;
  created_at: string;
}

File: backend/app/schemas/diagnostic.py
from typing import Optional, Dict, Any, List
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, Field, ConfigDict

class DiagnosticSessionCreate(BaseModel):
    asset_id: Optional[UUID] = None
    technician_id: Optional[UUID] = None
    user_question: str = Field(..., min_length=3)
    image_id: Optional[UUID] = None
    status: str = Field(default="pending", pattern="^(pending|processing|completed|failed)$")

class DiagnosticSessionResponse(BaseModel):
    id: UUID
    asset_id: Optional[UUID] = None
    technician_id: Optional[UUID] = None
    user_question: str
    image_id: Optional[UUID] = None
    status: str
    created_at: datetime
    completed_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

class DiagnosticResultCreate(BaseModel):
    diagnostic_session_id: UUID
    issue_summary: str
    identified_error_code: Optional[str] = None
    confidence: Optional[float] = None
    probable_causes: List[Dict[str, Any]] = Field(default_factory=list)
    recommended_steps: List[Dict[str, Any]] = Field(default_factory=list)
    required_tools: List[str] = Field(default_factory=list)
    safety_warnings: List[str] = Field(default_factory=list)
    citations: List[Dict[str, Any]] = Field(default_factory=list)
    model_name: Optional[str] = "gemini-1.5-pro"
    response_time_ms: Optional[int] = None

class DiagnosticResultResponse(DiagnosticResultCreate):
    id: UUID
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
